automation/jleechanorg_pr_automation/tests/test_automation_safety_limits.py

```
@@ -0,0 +1,461 @@
+            self.automation_manager.record_global_run()
+
+        # Simulate restart
+        new_manager = AutomationSafetyManager(self.test_dir)
+
+        # Should maintain run count
+        self.assertEqual(new_manager.get_global_runs(), 10)
+
+    # Matrix 6: Concurrent Access Safety
+    def test_concurrent_pr_attempts_thread_safe(self):
+        """RED: Concurrent PR attempts should be thread-safe"""
+        import threading
```

Ruff (0.14.5)

`import` should be at the top-level of a file

(PLC0415)